mutiny-core/src/vss.rs

```diff
@@ -0,0 +1,273 @@
+#![allow(dead_code)]
+use crate::auth::MutinyAuthClient;
+use crate::{error::MutinyError, logging::MutinyLogger};
+use aes::cipher::block_padding::Pkcs7;
+use aes::cipher::{BlockDecryptMut, BlockEncryptMut, KeyIvInit};
+use aes::Aes256;
+use anyhow::anyhow;
+use bitcoin::secp256k1;
+use bitcoin::secp256k1::SecretKey;
+use cbc::{Decryptor, Encryptor};
+use lightning::log_error;
+use lightning::util::logger::*;
+use reqwest::{Method, Url};
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+use std::sync::Arc;
+
+type Aes256CbcEnc = Encryptor<Aes256>;
+type Aes256CbcDec = Decryptor<Aes256>;
+
+pub struct MutinyVssClient {
+    auth_client: Arc<MutinyAuthClient>,
+    url: String,
+    encryption_key: SecretKey,
+    logger: Arc<MutinyLogger>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct VssKeyValueItem {
+    pub key: String,
+    pub value: Option<Value>,
+    pub version: u32,
+}
+
+impl VssKeyValueItem {
+    /// Encrypts the value of the item using the encryption key
+    /// and returns an encrypted version of the item
+    pub(crate) fn encrypt(self, encryption_key: &SecretKey) -> EncryptedVssKeyValueItem {
+        // should we handle this unwrap better?
+        let bytes = self.value.unwrap().to_string().into_bytes();
+        let iv: [u8; 16] = secp256k1::rand::random();
+
+        let cipher = Aes256CbcEnc::new(&encryption_key.secret_bytes().into(), &iv.into());
+        let mut encrypted: Vec<u8> = cipher.encrypt_padded_vec_mut::<Pkcs7>(&bytes);
+        encrypted.extend(iv);
+        let encrypted_value = base64::encode(encrypted);
```

Comment:
I think we're now using this encryption mechanism for here and SCB. Can we put this in `encrypt.rs`?

Reply:
done, might be able to unify them between that and the password encryption too